sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,139 @@
+package io.sentry.android.okhttp
+
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import io.sentry.TypeCheckHint
+import io.sentry.util.UrlUtils
+import okhttp3.Call
+import okhttp3.Response
+import java.net.URL
+
+private val uuidRegex by lazy { Regex("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}") }
+
+internal class SentryOkHttpEvent(private val hub: IHub, private val call: Call) {
+    private val eventSpans: MutableMap<String, ISpan> = HashMap()
+    private val breadcrumb: Breadcrumb
+    internal val callRootSpan: ISpan?
+    private var response: Response? = null
+
+    init {
+        val urlDetails = UrlUtils.parse(call.request().url.toString())
+        val url = urlDetails.urlOrFallback
+        val trimmedUrl: String = trimUrl(url)
+        val host: String = call.request().url.host
+        val encodedPath: String = call.request().url.encodedPath
+        val method: String = call.request().method
+
+        // We start the call span that will contain all the others
+        callRootSpan = hub.span?.startChild("http.client", "$method $url")
+
+        // We setup a breadcrumb with all meaningful data
+        breadcrumb = Breadcrumb.http(url, method)
+        breadcrumb.setData("url", url)
```

Comment:
`m`: I don't remember by heart if Relay scrubs PII from the `data` bag, but i'd double-check with the ingest team or somewhere develop docs maybe

Reply:
Why do we set `url` again?
`Breadcrumb.http` already does it.